google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardDeployTask.java

```diff
@@ -37,16 +39,27 @@
 
   private AppEngineDeploy deploy;
   private AppEngineStandardStage stageStandard;
+  private boolean isFlexCompat;
 
+  /**
+   * @param isFlexCompat does not change any behavior of actual deployment. Provided solely
+   *     for the purpose of Analytics usage reporting.
+   */
   public AppEngineStandardDeployTask(
       @NotNull AppEngineDeploy deploy,
-      @NotNull AppEngineStandardStage stageStandard) {
+      @NotNull AppEngineStandardStage stageStandard,
+      boolean isFlexCompat) {
     this.deploy = deploy;
     this.stageStandard = stageStandard;
+    this.isFlexCompat = isFlexCompat;
   }
 
   @Override
   public void execute(ProcessStartListener startListener) {
+    UsageTrackerProvider.getInstance()
+        .trackEvent(GctTracking.CATEGORY, GctTracking.APP_ENGINE_DEPLOY,
+            isFlexCompat ? "flex.compat" : "standard", null);
```

Comment:
Let's interpret that as "a standard app deployed to the flex env that is compatible with the standard". I think it's fine since this is just a label in Analytics and not a strong type in code.